File: 1916-find-center-of-star-graph/find-center-of-star-graph.cpp
class Solution {
public:
    int findCenter(vector<vector<int>>& edges) {
        int  maxi=INT_MIN;
        for(int i=0;i<edges.size();i++){
            maxi=max({maxi,edges[i][0],edges[i][1]});
        }
        vector<vector<int>>adj(maxi+1);
        for(auto i : edges){
            adj[i[0]].push_back(i[1]);
            adj[i[1]].push_back(i[0]);
        }
        int n=adj.size();
        for(int i=1;i<=n;i++){
            int c=0;
            
            if(adj[i].size()==maxi-1){
                return i;
            }
        }
        return -1;
    }
};

File: 1916-find-center-of-star-graph/find-center-of-star-graph.java
class Solution {
    public int findCenter(int[][] edges) {
        int common=0;
        for(int i=0;i<edges.length-1;i++)
        {
            if((edges[i][0]==edges[i+1][0])||(edges[i][0]==edges[i+1][1]))
                common=edges[i][0];
            else if((edges[i][1]==edges[i+1][0])||(edges[i][1]==edges[i+1][1]))
                common=edges[i][1];
        }
        return common;
    }
}